<launch>
  <arg name="param_path" default="$(find-pkg-share surround_obstacle_checker)/config/surround_obstacle_checker.param.yaml" />

  <arg name="input_trajectory" default="input/trajectory" />
  <arg name="input_objects" default="/perception/object_recognition/objects" />
  <arg name="input_odometry" default="/localization/kinematic_state" />
  <arg name="input_pointcloud" default="/perception/obstacle_segmentation/pointcloud" />
  <arg name="output_trajectory" default="output/trajectory" />

  <node pkg="surround_obstacle_checker" exec="surround_obstacle_checker_node" name="surround_obstacle_checker" output="screen">
    <param from="$(var param_path)" />
    <remap from="~/output/no_start_reason" to="/planning/scenario_planning/status/no_start_reason" />
    <remap from="~/output/stop_reasons" to="/planning/scenario_planning/status/stop_reasons" />
    <remap from="~/output/trajectory" to="$(var output_trajectory)" />
    <remap from="~/input/pointcloud" to="$(var input_pointcloud)" />
    <remap from="~/input/objects" to="$(var input_objects)" />
    <remap from="~/input/odometry" to="$(var input_odometry)" />
    <remap from="~/input/trajectory" to="$(var input_trajectory)" />
  </node>

</launch>
